Views and explores for numbers that matter mobile

acmiyaguchi opened the pull request:
[DS-1544](https://jira.mozilla.com/browse/DS-1544). See [this document](https://docs.google.com/document/d/1fRdgPRJK8sNz_5_DicxY49TO9Q7I3QS6LpyIgvGIbns/edit#) for details about various data sources.

This contains a view/explore for viewing the mobile funnel, alongside a dashboard. There are a few minor areas of adjustment that need to be made, but this should encapsulate a large amount of the functionality needed.

I have not added focus/klar since they depend on legacy telemetry. It should be possible to add if it's organized within the non_desktop_clients_last_seen table appropriately.

acmiyaguchi commented:
@fbertsch Does the dashboard not show up in the shared folder, even when development mode is on? I'd like to edit the dashboard text, but there doesn't seem to be an easy way to do so without copying over the the generated dashboard code from the one in my personal folder.

fbertsch reviewed (commented):
Just a few initial comments. Realizing it's probably impossible to get this in bigquery_etl.

acmiyaguchi commented on duet/views/mobile_android_country.view.lkml:
```diff
@@ -0,0 +1,226 @@
+    }
+  }
+
+  dimension: join_field {
```
This is necessary for the merged results between the current period and the previous period to get a percent change.

acmiyaguchi commented on duet/views/mobile_android_country.view.lkml:
```diff
@@ -0,0 +1,226 @@
+          sum(first_seen) as first_seen,
+          sum(activated) as activated
+      from play_store_retained
+      right join play_store_installs
```
This is leftover from my previous queries where the installs and last seen tables had dates that were a superset of the retained table. In practice, I've filtered down these tables so we can actually just do an inner join.

acmiyaguchi commented on duet/views/mobile_android_country.view.lkml:
```diff
@@ -0,0 +1,226 @@
+              first_seen_date as submission_date,
+              coalesce(play_store_countries.country, "OTHER") as country,
+              count(distinct client_id) as first_seen,
+              count(distinct case when coalesce(BIT_COUNT(days_seen_bits), 0) >= 5 then client_id end) as activated
```
I was using the 5/7 day activation that desktop was using in numbers that matter. I will confirm the definition with @ksiegler1 though, I didn't look too closely at using the proper definition.

ksiegler1 replied:
Using 5/7 day activation works for now, it's unclear whether 2/7 activation is going to be relevant to report on. George was working on mobile activation criteria before he went on pat leave but I don't believe one was settled on.

acmiyaguchi commented on duet/views/mobile_android_country.view.lkml:
```diff
@@ -0,0 +1,226 @@
+view: mobile_android_country {
+  derived_table: {
+    sql: with last_updated as (
```
Swapping between apps is done through an app id parameter and seems to work okay. I'm not sure how this would work in the context of a bigquery-etl view because of all the different parameterizations (e.g. how big should the current aggregation window be, which tables should we actually be using to count first seen clients, etc).

acmiyaguchi commented on duet/views/mobile_android_country.view.lkml:
```diff
@@ -0,0 +1,226 @@
+    sql: ${TABLE}.latest_date ;;
+  }
+
+  dimension: country {
```
I did this for the main geographical bucketing value, which is joined using the country dimension.

acmiyaguchi commented:
I fixed an issue with choosing the end-date, which is 7 days before today. I was using the play store updated date which can update sooner than 7 days.

I also joined the aggregates against the country buckets explore, which includes things like tier-1 and non-tier-1 countries. This means the sum of rows can be more than the original aggregates, so I force this this as a filter parameter.

Finally, I added focus/klar to the dashboard. I found an issue with the core_clients_last_seen table where the last updated values for focus/klar is 2021-04-01, so there is something wrong with the ETL there. On the other hand, it does seem to work without serious issues in the dashboard otherwise.

acmiyaguchi commented:
![image](https://user-images.githubusercontent.com/3304040/119064189-a6830b80-b98f-11eb-9f08-f68d4246a49e.png)

ksiegler1 reviewed (approved):
Mobile stakeholders will likely be interested in Days of Use as well, but I think that's something we can push until a little later.

acmiyaguchi commented:
I'm going to merge this in today so it can be shared on Monday. This should be a good starting point. It could use with some documentation about the subtly of the Play Store.

fbertsch commented:
@ksiegler1 @acmiyaguchi it would be good to use a `first_seen` dataset based on the activation ping. Those numbers for new users are going to fit in this funnel much better than the counts from the `last_seen` dataset; they should match roughly what the play store tells us (not accounting for non-play store installs). The activation ping table is not too large; we could probably re-materialize it from all of history every day for now.

We should probably change the name - `Mobile Android Country` is not descriptive of what it contains. Maybe `Android Acquisition Funnel`?

Should `join_field`, `play_store_updated`, and `latest_date` be hidden?

Consider also changing the app_name titles to e.g. `Firefox Android Release`, `Firefox Android Beta`, `Firefox Android Nightly`, etc.

acmiyaguchi commented:
Merging this in will make it easier for @ksiegler1 to eventually edit the dashboard. I filed https://jira.mozilla.com/browse/DS-1605 regarding the activation ping.

I don't know if `Android Acquisition Funnel` is descriptive enough, since there are other modes of acquisition (e.g. by attribution). My reasoning for the name is that this is the mobile funnel for android broken down by country. Maybe `Android Acquisition Funnel by Country`, if looker allows for a name that isn't derived from the file. 

`join_field` is necessary in the explore for merged results, `play_store_updated` and `latest_date` are both exposed in the dashboard because I think it's valuable to know the period the dashboard is being used in.  

Labels are probably a good idea, I'll make a follow up for that (though I am working on building an iOS equivalent of this dashboard atm).
